### PR TITLE
[ZEPPELIN-5755] Support Spark 3.3

### PR DESCRIPTION
**Pull request by zjffdu**
### What is this PR for?

Trivial PR to support spark 3.3
* Add profile spark-3.3
* Add integration tests for spark-3.3
* Due to api change, sqlContext in pyspark is set via different api in spark 3.3

### What type of PR is it?

Feature

### Todos
* [ ] - Task

### What is the Jira issue?
* https://issues.apache.org/jira/startup.jsp?returnTo=%2Fbrowse%2FZEPPELIN-5755
* 
### How should this be tested?
* CI pass

### Questions:
* Does the licenses files need to update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: .github/workflows/core.yml [hunk 1]
```diff
@@ -395,6 +395,14 @@ jobs:
         run: |
           rm -rf spark/interpreter/metastore_db
           ./mvnw test -DskipRat -pl spark-submit,spark/interpreter -Pspark-3.2 -Pspark-scala-2.13 -Phadoop2 -Pintegration -B -DfailIfNoTests=false
+      - name: run spark-3.3 tests with scala-2.12 and python-${{ matrix.python }}
+        run: |
+          rm -rf spark/interpreter/metastore_db
+          ./mvnw test -DskipRat -pl spark-submit,spark/interpreter -Pspark-3.3 -Pspark-scala-2.12 -Phadoop2 -Pintegration -B -DfailIfNoTests=false
```
Comment (Reamer) [inline .github/workflows/core.yml:401]: Can we test with Hadoop 3? Hadoop 3 has been available for a while now.

Comment (zjffdu) [reply .github/workflows/core.yml]: Done

**Comment by zjffdu** [comment]
Will merge if no more comment

**Review by Reamer — APPROVED** [review]
LGTM

**Comment by zjffdu** [comment]
@shakraz Spark 3.3 has not been released when Zeppelin 0.10.1 is released

**Pull request merged**